record_archiver/models/record_lifespan.py

```diff
@@ -0,0 +1,111 @@
+
+# Copyright 2015-2016 Yannick Vaucher (Camptocamp SA)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+import logging
+
+from datetime import datetime
+from dateutil.relativedelta import relativedelta
+
+from odoo import _, api, exceptions, fields, models
+
+_logger = logging.getLogger(__name__)
+
+
+class RecordLifespan(models.Model):
+    """Configure records lifespans per model.
+
+    After the lifespan is expired (compared to the `write_date` of the
+    records), the records are deactivated.
+    """
+    _name = 'record.lifespan'
+    _order = 'model'
+
+    model_id = fields.Many2one(
+        'ir.model',
+        string='Model',
+        required=True,
+        domain=[('has_an_active_field', '=', True)],
+    )
+    model = fields.Char(
+        related='model_id.model',
+        readonly=True,
+        string='Model Name',
+        store=True,
```

Comment:
passing by a second time... I don't think we need to store this field. 
1st of all we don't use it, 2nd we can easily search records using `('model_id.model', '=', 'foo.bar')`.
Am I wrong?

Reply:
Could be that initial purpose was to index `record.lifespan` records by model
nevertheless, I agree, and it looks just fine w/o that field